test-ts/contracts/PrecompileCaller.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+// Calls GA CENNZ precompile
+contract PrecompileCaller {
+    // derived CENNZ token address on testnets (AssetId 16000)
+    // CcccccCc (prefix) + 00003e80 (assetId) + padding
+    // run through web3.utils.toChecksumAddress(..)
+    address cennz = 0xcCccccCc00003E80000000000000000000000000;
+
+    function balanceOfProxy(address who) public view returns (uint256) {
+        (bool success, bytes memory returnData) = cennz.staticcall(abi.encodeWithSignature("balanceOf(address)", who));
+        assembly {
+            if eq(success, 0) {
+                revert(add(returnData, 0x20), returndatasize())
+            }
+        }
+        return abi.decode(returnData, (uint256));
+    }
+
+    // transfer CENNZ from caller using the CENNZ precompile address w ERC20 abi
+    function takeCENNZ(uint256 amount) external {
+        (bool success, bytes memory returnData) = cennz.call(abi.encodeWithSignature("transferFrom(address,address,uint256)", msg.sender, address(this), amount));
```

Comment:
have found this kind of code causes a revent when used with precompile.
still investigating the cause
```suggestion
       IERC20(cennz).transferFrom(msg.sender, address(this), amount);
```

Reply:
`Contract.function()` calls insert an EXTCODESIZE check which fails as precompiles have 0 codesize.
this check is removed in solidity 8.13 (https://github.com/ethereum/solidity/issues/12204)
for backwards compatibility it could be worth responding to the EXTCODESIZE thing

Reply:
frontier pallet_evm has `AccountStorages` which can  be set to non-zero for precompiles to pass this check